cms/static/js/views/pages/container.js

```diff
@@ -843,6 +949,30 @@ function($, _, Backbone, gettext, BasePage,
                 placeholderElement;
 
             placeholderElement = $placeholderEl.insertAfter(xblockElement);
+
+            if (this.options.isIframeEmbed) {
+                try {
+                    window.parent.postMessage(
+                        {
+                            type: 'scrollToXBlock',
+                            message: 'Scroll to XBlock',
+                            payload: { scrollOffset: xblockElement.height() }
+                        }, document.referrer
+                    );
+                } catch (e) {
+                    console.error(e);
+                }
+                return window.addEventListener(
+                    'message',
+                    (event) => {
+                        if (event.data && event.data.type === 'completeXBlockDuplicating') {
+                            return self.onNewXBlock(placeholderElement, null, true, event.data.payload);
+                        }
+                    },
+                    { once: true }
+                );
```

Comment:
Hmm, sorry, now that I look at this again, I realize that `{ once: true }` will be a problem. If an unrelated `message` event comes in before `completeXBlockDuplicating`, the listener will be called for that other message, and do nothing. But then the event handler will be unregistered. Then, when the `completeXBlockDuplicating` message comes in, nothing will happen.

I think you need some other approach to ensure that the event listener is only called once, either by unregistering itself within the `if (event.data && event.data.type === 'completeXBlockDuplicating') {` section, or add this event listener during the overall init and have it always listening rather than temporarily registering it here.

Reply:
Looks good to me - thanks!